### ISSUE TITLE
Bug: No optimizer.zero_grad in training function

### ISSUE THREAD

**Issue by venondev** [braindatalab/gecobench#62]
During the learning rate sweep I noticed, we did not call `optimizer.zero_grad()` in the training loop.

For the one_layer_attention model, it actually seems to improve the learning. Without it, the classifier achieves validiation accuracies of around 80% and with resetting the gradients it does not seem to learn anything and is stuck at 50%.

See PR: https://github.com/braindatalab/xai-nlp-benchmark/pull/57

With `optimizer.zero_grad()`:

![image](https://github.com/braindatalab/xai-nlp-benchmark/assets/22524997/fa0f6577-bdd8-4b6b-b1fa-4cd8b79989f2)

Without `optimizer.zero_grad()`

![image](https://github.com/braindatalab/xai-nlp-benchmark/assets/22524997/605f3343-bd29-46db-9e66-3e8e3b93ec45)

Both with different learning rates.
